mycity/mycity/intents/grocery_store_intent.py

```diff
@@ -0,0 +1,137 @@
+"""
+Grocery Store Intent
+"""
+import mycity.utilities.gis_utils as gis_utils
+import mycity.utilities.datetime_utils as date
+import logging
+from mycity.mycity_response_data_model import MyCityResponseDataModel
+from mycity.intents.intent_constants import CURRENT_ADDRESS_KEY
+from . import intent_constants
+from streetaddress import StreetAddressParser
+from .custom_errors import \
+    InvalidAddressError, BadAPIResponse, MultipleAddressError
+import mycity.intents.speech_constants.food_truck_intent as speech_constants
+from mycity.intents.user_address_intent import clear_address_from_mycity_object
+
+logger = logging.getLogger(__name__)
+
+BASE_URL = 'https://services.arcgis.com/sFnw0xNflSi8J0uh/ArcGIS/rest/' \
+           'services/Supermarkets_GroceryStores/FeatureServer/0'
+QUERY = {
+    "distance": "1",
+    "inSR": "4326",
+    "outSR": "4326",
+    "f": "json",
+    "outfields": "*",
+    "units": "esriSRUnit_StatuteMile",
+    "geometryType": "esriGeometryPoint"
+}
+DAY = date.get_day()
+CARD_TITLE = "Grocery Stores"
+
+
+def add_response_text(features):
+    response = ''
+    for t in features:
+        response += f"{t['attributes']['Store']}" \
+            f" in {t['attributes']['Neighborho']} is located at " \
+            f"{t['attributes']['Address']}. "
+    return response
+
+
+def get_grocery_grocery_stores(address_of_interest):
+    """
+    Gets the grocery store location within a mile from a given address
+    :return: a list of features with unique grocery stores
+    """
+
+    formatted_address = '{x_coordinate}, {y_coordinate}'.format(
+        x_coordinate=address_of_interest['x'],
+        y_coordinate=address_of_interest['y']
+    )
+    QUERY["geometry"] = formatted_address
+
+    grocery_stores = gis_utils.get_features_from_feature_server(BASE_URL, QUERY)
+    return grocery_stores
+
+
+def get_nearby_grocery_stores(mycity_request):
+    """
+    Get grocery stores near an address
+
+    :param mycity_request: MyCityRequestDataModel object
+    :return: MyCityResponseObject
+    """
+    mycity_response = MyCityResponseDataModel()
+
+    # Get current address location
+    if CURRENT_ADDRESS_KEY in mycity_request.session_attributes:
+        current_address = \
+            mycity_request.session_attributes[CURRENT_ADDRESS_KEY]
+
+        # Parsing street address using street-address package
+        address_parser = StreetAddressParser()
+        a = address_parser.parse(current_address)
+        address = str(a["house"]) + " " + str(a["street_name"]) + " " \
+                  + str(a["street_type"])
+
+        # Parsing zip code
+        zip_code = str(a["other"]).zfill(5) if a["other"] else None
+        zip_code_key = intent_constants.ZIP_CODE_KEY
+        if zip_code is None and zip_code_key in \
+                mycity_request.session_attributes:
+            zip_code = mycity_request.session_attributes[zip_code_key]
```

Comment:
It doesn't look like we use zip code except in an error message. We should probably just get rid of it here or use it to get a more accurate result from `geocode_address`

Reply:
Removed the zip code